src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_DIMENSION = 10;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+        switch (figureType) {
+            case 0:
+                return new Square((random.nextInt(MAX_DIMENSION) + 1) * MAX_DIMENSION, color);
+            case 1:
+                return new RightTriangle((random.nextInt(MAX_DIMENSION) + 1) * MAX_DIMENSION,
+                        (random.nextInt(MAX_DIMENSION) + 1) * MAX_DIMENSION, color);

```

Comment:
Similar to the previous comment, avoid passing expressions directly in the constructor. Declare variables for the dimension values before passing them to the constructor.